lib/gears/color.lua

```diff
@@ -59,11 +59,17 @@ local pattern_cache
 -- Max 4 chars per channel.
 --
 -- @param col The color to parse
--- @treturn table 4 values representing color in RGBA format (each of them in
--- [0, 1] range) or nil if input is incorrect.
+-- @treturn number between 0 and 1 for the 'red' value (1st channel)
+-- @treturn number between 0 and 1 for the 'green' value (2nd channel)
+-- @treturn number between 0 and 1 for the 'blue' value (3rd channel)
+-- @treturn number between 0 and 1 for the 'opacity' value (4th channel)
+-- if the incoming color code only has 3 values (only rgb, not opacity) the 4th
+-- return value is 1.
+-- @treturn nil if input is incorrect
```

Comment:
Hm... I'm not sure whether this is a good way to present the nil value returned in case of error. I don't know how to represent that better, tho. So let's keep it this way for now :shrug:

Reply:
I know what you mean - there should be an ldoc tag to indicate what a function returns in case of unexpected input.

Would you prefer to leave “nil in case of invalid input” out of the doc entirely? If someone calls parse_color with the wrong value, the error message will indicate that they’re trying to use a nil value anyway.

Reply:
Hm, yes, you are right. Let's remove it!

> there should be an ldoc tag to indicate what a function returns in case of unexpected input.

We could do it from our `config.ld` thanks to @Elv13 magic. But this would add a layer of complexity into account in the documentation HTML template and, as a whole, in our user facing API. I don't feel like this is necessary to implement such ldoc tag for 99% of the scenarios where a function is just supposed to return some value and if it fails because of user giving wrong values, it fails anyway.

Hm... Now that I think of it, maybe instead of returning nil, this function should simply invoke error (https://www.lua.org/manual/5.1/manual.html#pdf-error)

Anyway, I think this is out of the scope for this PR.

Reply:
> Hm... Now that I think of it, maybe instead of returning nil, this function should simply invoke error

Awesome's API never panics. So we shouldn't start doing that somewhere out of the blue.

Even the fact that this prints an error on invalid input is somewhat rare.